src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForLoopStatement.cs

```diff
@@ -3001,6 +3001,202 @@ static bool GetBool(out int x)
             VerifyOperationTreeAndDiagnosticsForTest<ForStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
         }
 
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(23821, "https://github.com/dotnet/roslyn/issues/23821")]
+        public void IForLoopStatement_MultipleIncrements_02()
```

Comment:
Consider using one of the simpler failing tests from the bug. Alternatively, consider making adding IOp verification to all of the tests that Aleksey listed in the bug.

Reply:
Also consider constraining the bind to just the last two statements of the for, as that's the part that was causing the exception previously

---
In reply to: [198630711](https://github.com/dotnet/roslyn/pull/28149#discussion_r198630711) [](ancestors = 198630711)

Reply:
Personally, I was in two minds on whether we should even add any new test here. We already have an assert in the test visitor and the test hook will catch it for all the mentioned tests in the bug if we were to regress it. I am not sure how to minimize the test, while guaranteeing that it would have failed for the original binder code where we had a syntax list as the node. I think it is fine to just have this single regression test for now, which we know for sure failed originally.

Reply:
Well, there were a bunch of failing tests in the original bug, and some of them were more minimal than this.

Reply:
We don't have the IOperation test hook running in automation yet, so I think we should keep the test. I was just suggesting a way to reduce the wall of text.

Reply:
Issue is there is no failing test right now, and we have no good way to come up with a regression test as the faulty code has been fixed long back. I feel asserts in both our test visitors are fine and guards against the original issue.